complex/ft_test.c

```diff
@@ -205,15 +218,10 @@ static void ft_format_iov_random(struct iovec *iov, size_t cnt, char *buf,
 	offset = 0;
 	for (i = 0; i < cnt; i++) {
 		/* If last IOV then use remaining data. */
-		if (i == (cnt - 1)) {
+		if (i == (cnt - 1))
 			weight = len;
-		} else {
-			/* Get a weight between 1 and the remaining length minus
-			 * the remaining IOV count. This is so we can reserve at
-			 * least a length of 1 for every IOV.
-			 */
-			weight = (rand() % (len - (cnt - i))) + 1;
-		}
```

Comment:
The other iov formatter methods allow for an iov to have a count/length of 0 except for this one so when the number of iovs > total elements being transferred, it ends up diving by 0. This change is to avoid a floating point exception by modifying the range it's selecting between. It wasn't a problem before because the iov_count never exceeded the total length of the message, but the iocs work with counts, not lengths, so they end up being a lot smaller than the lengths.